pkg/controller/common.go

```diff
@@ -136,3 +409,65 @@ func cleanUpAzVolumeAttachments(ctx context.Context, client client.Client, azCli
 func shouldCleanUp(attachment v1alpha1.AzVolumeAttachment, mode cleanUpMode) bool {
 	return mode == all || (attachment.Spec.RequestedRole == v1alpha1.PrimaryRole && mode == primaryOnly) || (attachment.Spec.RequestedRole == v1alpha1.ReplicaRole && mode == replicaOnly)
 }
+
+func isAttached(attachment *v1alpha1.AzVolumeAttachment) bool {
+	return attachment != nil && attachment.Status.Detail != nil && attachment.Status.Detail.PublishContext != nil
+}
+
+func isCreated(volume *v1alpha1.AzVolume) bool {
+	return volume != nil && volume.Status.Detail != nil && volume.Status.Detail.ResponseObject != nil
+}
+
+func deletionRequested(objectMeta *metav1.ObjectMeta) bool {
+	if objectMeta == nil {
+		return false
+	}
+	now := metav1.Now()
+	return objectMeta.DeletionTimestamp.Before(&now)
+}
+
+func finalizerExists(finalizers []string, finalizerName string) bool {
+	for _, finalizer := range finalizers {
+		if finalizer == finalizerName {
+			return true
+		}
+	}
+	return false
+}
+
+func labelExists(labels map[string]string, label string) bool {
+	if labels != nil {
+		_, ok := labels[label]
+		return ok
+	}
+	return false
+}
+
+func createLabelRequirements(label, value string) (*labels.Requirement, error) {
+	req, err := labels.NewRequirement(label, selection.Equals, []string{value})
+	if err != nil {
+		return nil, err
+	}
+	if req == nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Unable to create Requirement to for label key : (%s) and label value: (%s)", label, value))
+	}
+	return req, nil
+}
+
+func getNamespacedName(namespace, name string) string {
```

Comment:
np: This function name is somewhat misleading in that it does not return a `types.NamespacedName`. Consider `getQualifiedName` instead.